### PR TITLE
8280965: Tests com/sun/net/httpserver/simpleserver fail with FileSystemException on Windows 11

### PR DESCRIPTION
**Pull request by FrauBoes**
This change updates the tests in question to handle the case where sym link creation is not supported, as observed on Windows 11 where additional privileges are required to create sym links.

Testing: tier 1-3 across platforms, plus tests in question on Windows 11 specifically

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280965](https://bugs.openjdk.java.net/browse/JDK-8280965): Tests com/sun/net/httpserver/simpleserver fail with FileSystemException on Windows 11

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7335/head:pull/7335` \
`$ git checkout pull/7335`

Update a local copy of the PR: \
`$ git checkout pull/7335` \
`$ git pull https://git.openjdk.java.net/jdk pull/7335/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7335`

View PR using the GUI difftool: \
`$ git pr show -t 7335`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7335.diff">https://git.openjdk.java.net/jdk/pull/7335.diff</a>

</details>

### CODE REVIEW

File: test/jdk/com/sun/net/httpserver/simpleserver/SimpleFileServerTest.java [hunk 1]
```diff
@@ -52,6 +52,7 @@
 import jdk.test.lib.Platform;
 import jdk.test.lib.net.URIBuilder;
 import jdk.test.lib.util.FileUtils;
+import jtreg.SkippedException;
```
Comment (dfuch) [inline test/jdk/com/sun/net/httpserver/simpleserver/SimpleFileServerTest.java:55]: You should use the testng exception instead - testng doesn't recognize the jtreg exception

File: test/jdk/com/sun/net/httpserver/simpleserver/CustomFileSystemTest.java [hunk 2]
```diff
@@ -71,6 +71,7 @@
 import com.sun.net.httpserver.SimpleFileServer.OutputLevel;
 import jdk.test.lib.Platform;
 import jdk.test.lib.net.URIBuilder;
+import jtreg.SkippedException;
```
Comment (dfuch) [inline test/jdk/com/sun/net/httpserver/simpleserver/CustomFileSystemTest.java:74]: Same remark here: 
You should use the testng exception instead - testng doesn't recognize the jtreg exception

File: test/jdk/com/sun/net/httpserver/simpleserver/CustomFileSystemTest.java [hunk 3]
```diff
@@ -422,19 +424,31 @@ public void testSymlinkSegmentGET() throws Exception {
...
+                assertEquals(response.statusCode(), 404);
+                assertEquals(response.headers().firstValue("content-length").get(), expectedLength);
+                assertEquals(response.body(), expectedBody);
+            } finally {
+                server.stop(0);
+            }
+        }
+    }
+
+    private boolean createSymLink(Path symlink, Path target) {
```
Comment (Michael-Mc-Mahon) [inline test/jdk/com/sun/net/httpserver/simpleserver/CustomFileSystemTest.java:444]: Seems like createSymLink should actually return true or false, or if it throws an exception in one case, then it should be void, in which case you wouldn't need the if statement in the calling code

Comment (FrauBoes) [reply test/jdk/com/sun/net/httpserver/simpleserver/CustomFileSystemTest.java:444]: Good catch, that makes for cleaner code - I'll refactor.

**Review by dfuch — APPROVED** [review]
Changes look good to me!

**Pull request closed without merging**